### PR TITLE
[GOSCALR] SCALRCORE-31981 Provider > Add support SSH key into the Scalr Provider

### PR DESCRIPTION
**Pull request by RomanMytsko**

### CODE REVIEW

File: ssh_key_link.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	SSHKeyID string `json:"ssh-key"`
+}
+
+// Create creates a SSH key workspace link.
+func (s *sshKeysLinks) Create(ctx context.Context, workspaceID string, sshKeyID string) (*SSHKeysLink, error) {
+	urlPath := fmt.Sprintf("workspaces/%s/ssh-key-links", url.QueryEscape(workspaceID))
+	linkOptions := SSHKeysLinkCreateOptions{
+		SSHKeyID: sshKeyID,
+	}
+	req, err := s.client.newJsonRequest("POST", urlPath, linkOptions)
```
Comment (petroprotsakh) [inline ssh_key_link.go:37]: Something weird here, why do we create a json request, but later adding header for `vnd.api+json`?
```suggestion
	req, err := s.client.newRequest("POST", urlPath, linkOptions)
```

File: ssh_key_link.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/vnd.api+json")
+
+	link := &SSHKeysLink{}
```
Comment (petroprotsakh) [inline ssh_key_link.go:43]: That's not accurate; API spec defines that the response of create link operation is a workspace resource.

Comment (RomanMytsko) [reply ssh_key_link.go]: Fixed.

File: ssh_key_link.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	SSHKeyID string `json:"ssh-key"`
+}
+
+// Create creates a SSH key workspace link.
+func (s *sshKeysLinks) Create(ctx context.Context, workspaceID string, sshKeyID string) (*Workspace, error) {
+	urlPath := fmt.Sprintf("workspaces/%s/ssh-key-links", url.QueryEscape(workspaceID))
+	linkOptions := SSHKeysLinkCreateOptions{
+		SSHKeyID: sshKeyID,
+	}
+	req, err := s.client.newJsonRequest("POST", urlPath, linkOptions)
```
Comment (RomanMytsko) [inline ssh_key_link.go:37]: тут залишив `newJsonRequest`

**Review by petroprotsakh — APPROVED** [review]

**Pull request merged**